### PR TITLE
fix edit by area zoom

### PR DESCRIPTION
**Pull request by edgardoh**
Fixed in color zones, when edit by area is selected, zoomed in and nodes are edited.

### CODE REVIEW

**Comment by junkyardsparkle** [comment]
The y-offset issues I was having previously appear to be fixed in this branch. :)

File: src/iop/colorzones.c [hunk 1]
```diff
@@ -1239,15 +1224,19 @@ static gboolean _bottom_area_button_press_callback(GtkWidget *widget, GdkEventBu
 
 static int _sanity_check(const float x, const int selected, const int nodes, const dt_iop_colorzones_node_t *curve)
 {
-  int point_valid = 1;
+  int point_valid = TRUE;
```
Comment (TurboGit) [inline src/iop/colorzones.c]: should be a gboolean

**Comment by edgardoh** [comment]
> *@TurboGit* requested changes on this pull request.
>
> Just a minor comment. An int used as a boolean.
>
> A question, I found that a slight move of the luminence curve in strong
> mode gives very quickly artifacts. Do you reproduce that? As soon as I
> change for the smooth mode all is correct. It seems that the strong mode
> creates some cusp.
>
Can be, according to the comments on the code there's a mask used to avoid
artifacts, but is used only on hue, the lightness and chroma should work
the same. But the side effect of that mask is that it can produce a shift
on the destination hue, so sometimes you may not get the desired effect.
Having this two modes give the user the choice, if there's artifacts on the
image one can always change to the smooth method.
Actually I haven's seen any artifacts, but that may depend on the
definition of artifact, on very noisy images it brings the noise quite a
lot, that can be considered as artifacts?

------------------------------

**Comment by edgardoh** [comment]
Ready.

**Comment by TurboGit** [comment]
Thanks!

**Comment by TurboGit** [comment]
I'll open an issue with an image with artifacts just to be sure there is nothing broken.

**Comment by edgardoh** [comment]
If you're gonna be playing with it and you find the time check how nodes
are added when edit by area is selected. I don't use that mode so for me is
OK like it is, but you never know.

**Pull request merged**